### DISCUSSION TITLE
Is there a way to fail if I create an update statement without set?

### DISCUSSION THREAD

**Discussion by robstoll** [jOOQ/jOOQ#15743] [Q&A] [answered]
for instance, something like (kotlin code):
```kotlin
val rowsUpdated = context.update(TABLE).set(mapOf<Field<*>, Any>()).where(...).execute()
// rowsUpdated = 0
```
does not throw as I would expect in SQL when leaving out `SET ..` but simulates ~~as nothing was updated~~ as no rows was affected. Is there a config to change this behaviour?

**Comment by lukaseder** [comment, marked as answer]
> does not throw as I would expect in SQL when leaving out `SET ..`

SQL doesn't support an empty `SET` clause, but jOOQ does. The question is: What does it mean for an empty set clause to be passed? It could mean two things:

1. All rows matching the `WHERE` clause should be locked, but nothing should be changed
2. Nothing happens (this is what jOOQ opted for)

I would assume that if SQL supported the syntax, then 1) would be the correct approach, similar to running a `SELECT .. FOR UPDATE`, though the `UPDATE` would actually fire triggers.

You cannot change this behaviour,  but you can intercept such statements by implementing an `ExecuteListener` and checking `executeContext.query() instanceof QOM.Update u && u.$set().isEmpty()`

**Reply by robstoll** [reply]
great, thanks
